lib/superagent-mock.js

```diff
@@ -52,68 +48,25 @@ function mock (superagent, config, logger) {
   }();
 
   /**
-   * Override send function
-   */
-  Request.prototype.send = function (data) {
-    if (logEnabled) {
-      currentLog.data = data;
-    }
-
-    this.params = data;
-
-    return this;
-  };
-
-  /**
-   * Returns true if `obj` is an object.
-   */
-  function isObject(obj) {
-    return obj === Object(obj);
-  }
-
-  /**
-   * Override set function
-   */
-  Request.prototype.set = function (field, val) {
-    if (logEnabled && !isObject(field)) {
-      var headerPart = {};
-      headerPart[field] = val;
-      currentLog.headers = (currentLog.headers || []).concat(headerPart);
-    }
-
-    if (isObject(field)) {
-        for (var key in field) {
-          this.set(key, field[key]);
-        }
-    } else {
-        this.headers = this.headers || {};
-        this.headers[field] = val;
-    }
-
-    return this;
-  };
-
-  /**
    * Override end function
    */
   Request.prototype.end = function (fn) {
-    var path = this.url;
-    var querystring = '';
     var error = null;
-
-    if (this._query) {
-      querystring += this._query.join('&');
-    } else {
-      if (this.qs) {
-        querystring += qs.stringify(this.qs);
-      }
-      if (this.qsRaw) {
-        querystring += this.qsRaw.join('&');
-      }
+    var path;
+    var isNodeServer = this.hasOwnProperty('cookies');
+
+    if (isNodeServer) { // node server
```

Comment:
Server side and client side behaviours are different because of a different implementation of `_appendQueryString` in superagent, right?